frontend/src/api/tests/StreamApi.test.js

```diff
@@ -2,56 +2,163 @@ import axios from "axios";
+        status: "online"
+      },
+      input: null
+    }
+  },
+  outputChannel: {
+    id: 2,
+    channel: {
+      id: 4,
+      name: "Sample Sender Channel 2",
+      port: 20001
+    },
+    encoder: {
+      serialNumber: "d7TxFn7o",
+      device: {
+        serialNumber: "d7TxFn7o",
+        publicIpAddress: "0:0:0:0:0:0:0:1",
+        privateIpAddress: "127.0.0.1",
+        displayName: "sample_sender",
+        status: "online"
+      },
+      output: null
+    }
+  },
+  isRendezvous: false
 };
-const mockStreams = [mockSingleStream, { streamId: 2, streamName: "Test 2" }];
 
-const respAllStreams = {
-  target: {
-    streams: mockStreams
-  }
-};
-const respSingleStream = {
-  target: {
-    stream: mockSingleStream
-  }
+const axiosGetSecondStreamResponse = {
+  id: 2,
+  inputChannel: {
+    id: 2,
+    channel: {
+      id: 2,
+      name: "Sample Receiver Channel 2",
+      port: 20003
+    },
+    decoder: {
+      serialNumber: "z7VBn9aK",
+      device: {
+        serialNumber: "z7VBn9aK",
+        publicIpAddress: "0:0:0:0:0:0:0:1",
+        privateIpAddress: "127.0.0.1",
+        displayName: "sample_receiver",
+        status: "online"
+      },
+      input: null
+    }
+  },
+  outputChannel: {
+    id: 1,
+    channel: {
+      id: 3,
+      name: "Sample Sender Channel 1",
+      port: 20000
+    },
+    encoder: {
+      serialNumber: "d7TxFn7o",
+      device: {
+        serialNumber: "d7TxFn7o",
+        publicIpAddress: "0:0:0:0:0:0:0:1",
+        privateIpAddress: "127.0.0.1",
+        displayName: "sample_sender",
+        status: "online"
+      },
+      output: null
+    }
+  },
+  isRendezvous: false
 };
 
+const expectedFirstStreamRespone = new StreamInfo(
+  1,
+  convertToDataObject(axiosGetFirstStreamResponse.outputChannel.encoder),
+  convertToDataObject(axiosGetFirstStreamResponse.inputChannel.decoder),
+  ["Additional stream info goes here."]
+);
+const expectedSecondStreamRespone = new StreamInfo(
+  2,
+  convertToDataObject(axiosGetSecondStreamResponse.outputChannel.encoder),
+  convertToDataObject(axiosGetSecondStreamResponse.inputChannel.decoder),
+  ["Additional stream info goes here."]
+);
+
+const expectedAllStreamsResponse = [
+  expectedFirstStreamRespone,
+  expectedSecondStreamRespone
+];
+
 describe("Stream Api", () => {
   afterEach(() => {
     jest.clearAllMocks();
   });
 
   describe("getStream", () => {
-    it("should call axios.get and return a single stream's information", () => {
-      StreamApi.getStream(123);
+    it("should call axios.get and return a single stream's information", async () => {
+      axios.get.mockResolvedValue({ data: axiosGetFirstStreamResponse });
+      const response = await StreamApi.getStream(123);
 
       // check that callback was invoked with correct value
       expect(axios.get).toHaveBeenCalledWith(
         `http://localhost:8080/stream/123`
       );
+
+      expect(response).toEqual(expectedFirstStreamRespone);
```

Comment:
How come the response is equal to an object here and not a Promise?

Reply:
Because i `await`ed the return value of `getStream`.

Reply:
makes sense